python/pyarrow/tests/test_table.py

```diff
@@ -553,6 +553,34 @@ def test_recordbatch_dunder_init():
         pa.RecordBatch()
 
 
+def test_recordbatch_c_array_interface():
+    class BatchWrapper:
+        def __init__(self, batch):
+            self.batch = batch
+
+        def __arrow_c_array__(self, requested_type=None):
+            return self.batch.__arrow_c_array__(requested_type)
+
+    data = pa.record_batch([
+        pa.array([1, 2, 3])
+    ], names=['a'])
```

Comment:
Nit: make the type explicit for clarity?
```suggestion
    data = pa.record_batch([
        pa.array([1, 2, 3]), type=pa.int64()
    ], names=['a'])
```